### PR TITLE
Misc database sidebar updates

### PR DESCRIPTION
**Pull request by benkeen**
This contains various updates to the database sidebar, as per
Justin's feedback.
- Order and labels of buttons are changed.
- a few unclear icons (map reduce, Views) have been removed
- properly highlights what page you're on.
- Now allows us to select any nav item in the sidebar
regardless of nesting to highlight the nav item. This also
now works on full page refreshes.

Note this feature paves the way for the next thing I'll work
on: updating Views so that they retain the sidebar. Once
that's changed, it will need to clearly highlight what
View you're on in the sidebar.

### CODE REVIEW

File: app/addons/documents/assets/less/sidenav.less [hunk 1]
```diff
@@ -16,176 +16,142 @@
...
+      width: @sidebarWidth - 30px;
+      display: block;
+      padding: 10px 13px 10px 36px;
+      text-overflow: ellipsis;
+      overflow: hidden;
     }
   }
 }
+
+.index-group-header {
```
Comment (robertkowalski) [inline app/addons/documents/assets/less/sidenav.less]: maybe namespace it somehow, e.g by nesting and/or naming?

File: app/addons/documents/sidebar/sidebar.react.jsx [hunk 2]
```diff
@@ -380,24 +403,28 @@ function (app, FauxtonAPI, React, ReactDOM, Stores, Actions, Components, Documen
     },
 
     onChange: function () {
-      this.setState(this.getStoreState());
+      if (this.isMounted()) {
```
Comment (robertkowalski) [inline app/addons/documents/sidebar/sidebar.react.jsx]: cool, we should add those conditionals more often

File: app/addons/documents/sidebar/stores.js [hunk 3]
```diff
@@ -42,62 +54,89 @@ function (app, FauxtonAPI, ActionTypes) {
...
+    getToggledSections: function () {
+      return this._toggledSections;
+    },
 
+    getDatabaseName: function () {
+      if (this.isLoading()) {
+        return '';
+      }
       return this._database.safeID();
     },
```
Comment (robertkowalski) [inline app/addons/documents/sidebar/stores.js]: maybe make the model have an property called 

`this._database.safeId` which is a string, and until it is populated an empty string

or a variation:

`this._database.name`

don't like the fact that we pass the model to the view and the view calls getter on the model, if the view thinks it has finished loading.

**Comment by robertkowalski** [comment]
<3 the CSS refactor & also the new design.

the pink feels a bit strange, maybe Justin can give us a hint, he usually has really good ideas for colors

apart from the namespacing thing, +1

File: app/addons/documents/assets/less/sidenav.less (hunk 1 above)
Comment (benkeen) [inline app/addons/documents/assets/less/sidenav.less]: Fair point! But I'd argue it's okay, actually. It's a pretty descriptive (specific) class name and it's the namespacing that ends up getting us in trouble with deep selectors + difficulty overriding them later. I did a search over the code base to ensure uniqueness (I always do when adding new selectors).

**Comment by benkeen** [comment]
Yeah, let me run the colour by Justin one more time. I tweaked the pink a little to make it less awful but it could still use some love.

**Comment by benkeen** [comment]
Thanks @robertkowalski. Justin had a number of other small changes which I'll add shortly.

File: app/addons/documents/assets/less/sidenav.less (hunk 1 above)
Comment (benkeen) [inline app/addons/documents/assets/less/sidenav.less]: EDIT: I've namespaced this in .sidenav, just to get this ticket merged.

**Pull request merged**